src/util/log/Logger.cpp

```diff
@@ -65,6 +71,30 @@
 
 namespace util {
 
+namespace {
+
+class LoggerExceptionHandler {
+    std::reference_wrapper<util::prometheus::CounterInt> exceptionCounter_ =
+        PrometheusService::counterInt("logger_exceptions_total_number", util::prometheus::Labels{});
+
+public:
+    using result_type = void;
+
+    LoggerExceptionHandler()
+    {
+        // Check that prometheus is already initialized
+        PrometheusService::instance();
```

Comment:
This feels a bit like a hack :rage4:  
Maybe add a static member function that returns whether the instance exists or not without ASSERTing and use an ASSERT here and in `instance()` itself using the new function?